pkg/did/client.go

```diff
@@ -33,11 +33,11 @@ const (
 )
 
 type discovery interface {
-	GetEndpoints(domain string) ([]*endpoint.Endpoint, error)
+	GetEndpoints(domain string) ([]*endpoint.Endpoint, int, error)
```

Comment:
The return arg (numStakeholders) shouldn't be part of discovery. Discovery simply returns all possible endpoints and selection must use a policy to select from them.

Reply:
Then I need to refactor out part of discovery, so the client fetches the consortium (to have access to the policy), then the discovery service is provided the consortium and finds all the endpoints.

Reply:
I think the reason I wasn't having any success with the refactoring is that it will need me to implement part of config caching to make it work - otherwise I'm just creating a new discovery service then throwing it away, each call. The better solution would be to have the config cached, so the discovery service can be created once for the consortium, and use the cached config for its consortium-specific parameters.